File: java/link-message-client/src/main/java/link/message/client/content/complex/Param.java
package link.message.client.content.complex;

import link.message.client.utils.Guard;

public class Param {
	protected String key;
	protected String value;
	
	protected Param() {
		
	}
	
	public Param(String key, String value) {
		Guard.guardReqiredString(key,   "key must be set value.");
		Guard.guardReqiredString(value, "key must be set value.");
		
		this.key   = key;
		this.value = value;
	}

	public String getKey() {
		return key;
	}

	public void setKey(String key) {
		this.key = key;
	}

	public String getValue() {
		return value;
	}

	public void setValue(String value) {
		this.value = value;
	}

}

File: net/Link.Message.Client/content/complex/Param.cs
using Link.Message.Client.utils;

namespace Link.Message.Client.content.complex
{
	public class Param
	{
	    public string Key { get;set; }
	    public string Value { get; set; }

		protected internal Param()
		{

		}

		public Param(string key, string value)
		{
			Guard.GuardReqiredString(key, "key must be set value.");
			Guard.GuardReqiredString(value, "key must be set value.");

			this.Key = key;
			this.Value = value;
		}
	}
}